permissions_validators/src/private_blockchain/tests/revoke_and_grant.rs

```diff
@@ -64,7 +64,7 @@ fn add_register_domains_permission_denies_registering_domain_with_wrong_token()
 
     let mut alice = Account::new(alice_id.clone(), []).build();
     alice.add_permission(PermissionToken::new(
-        Name::from_str("incorrecttoken").expect("Valid"),
+        "incorrecttoken".parse().expect("Valid"),
```

Comment:
IMO `PermissionToken` could implement `FromStr` directly in some cases which would reduce the code to:
```rs
alice.add_permission(PermissionToken::from_str("incorrect_token"));
```

Reply:
Or 
```rust
PermissionToken::new("incorrect_token").expect("valid");
```

Rust dislikes hidden allocations, but constructors are allocating by definition.

Reply:
I'm not sure why to introduce a constructor like that. What if we already have `Name`? We can't move it to the `PermissionToken`, we should always deref it and clone

Reply:
> I'm not sure why to introduce a constructor like that. What if we already have Name? We can't move it to the PermissionToken, we should always deref it and clone

you are correct, this is the convention we use in `data_model`